guides/keras_nlp/fnet_classification.py

```diff
@@ -0,0 +1,380 @@
+    mask_zero=True,
+)(input_ids)
+
+x = keras_nlp.layers.FNetEncoder(intermediate_dim=INTERMEDIATE_DIM)(inputs=x)
+x = keras_nlp.layers.FNetEncoder(intermediate_dim=INTERMEDIATE_DIM)(inputs=x)
+x = keras_nlp.layers.FNetEncoder(intermediate_dim=INTERMEDIATE_DIM)(inputs=x)
+
+
+x = keras.layers.GlobalAveragePooling1D()(x)
+x = keras.layers.Dropout(0.1)(x)
+outputs = keras.layers.Dense(1, activation="sigmoid")(x)
+
+fnet_classifier = keras.Model(input_ids, outputs, name="fnet_classifier")
+
+"""
+## Training our Model
+
+We'll use accuracy to monitor training progress on the validation data.
+
+We will train our model for 3 epochs.
+"""
+fnet_classifier.summary()
+fnet_classifier.compile(
+    optimizer=keras.optimizers.Adam(learning_rate=0.001),
+    loss="binary_crossentropy",
+    metrics=["accuracy"],
+)
+fnet_classifier.fit(train_ds, epochs=EPOCHS, validation_data=val_ds)
+
+"""
+We obtain a train accuracy of around 70% and a validation accuracy of around
+71%. Moreover, for 3 epochs, it takes around 9 seconds to train the model (on Colab
+with a 16 GB Tesla T4 GPU).
+
+Let's calculate the test accuracy.
+"""
+fnet_classifier.evaluate(test_ds, batch_size=BATCH_SIZE)
+
+"""
+We obtain a test accuracy of around 72%.
+"""
+
+"""
+## Comparison with Transformer Model
+
+Let's compare our FNet Classifier model with a Transformer Classifier model. We
+keep all the parameters/hyperparameters the same. For example, we use three
+`TransformerEncoder` layers.
+
+We set the number of heads to 2.
+"""
+NUM_HEADS = 2
+input_ids = keras.Input(shape=(None,), dtype="int64", name="input_ids")
+
+
+x = keras_nlp.layers.TokenAndPositionEmbedding(
+    vocabulary_size=VOCAB_SIZE,
+    sequence_length=MAX_SEQUENCE_LENGTH,
+    embedding_dim=EMBED_DIM,
+    mask_zero=True,
+)(input_ids)
+
+x = keras_nlp.layers.TransformerEncoder(
+    intermediate_dim=INTERMEDIATE_DIM, num_heads=NUM_HEADS
+)(inputs=x)
+x = keras_nlp.layers.TransformerEncoder(
+    intermediate_dim=INTERMEDIATE_DIM, num_heads=NUM_HEADS
+)(inputs=x)
+x = keras_nlp.layers.TransformerEncoder(
+    intermediate_dim=INTERMEDIATE_DIM, num_heads=NUM_HEADS
+)(inputs=x)
+
+
+x = keras.layers.GlobalAveragePooling1D()(x)
+x = keras.layers.Dropout(0.1)(x)
+outputs = keras.layers.Dense(1, activation="sigmoid")(x)
+
+transformer_classifier = keras.Model(input_ids, outputs, name="transformer_classifier")
+
+
+transformer_classifier.summary()
+transformer_classifier.compile(
+    optimizer=keras.optimizers.Adam(learning_rate=0.001),
+    loss="binary_crossentropy",
+    metrics=["accuracy"],
+)
+transformer_classifier.fit(train_ds, epochs=EPOCHS, validation_data=val_ds)
+
+"""
+We obtain a train accuracy of around 78% and a validation accuracy of around
+72%. It takes around 14 seconds to train the model.
+
+Let's calculate the test accuracy.
+"""
+transformer_classifier.evaluate(test_ds, batch_size=BATCH_SIZE)
+
+"""
+We obtain a test accuracy of 73.62%.
+"""
+
+"""
+Let's make a table and compare the two models.
```

Comment:
I would state this point a little more clearly--We can see that FNet significantly speeds up our run time, with only a small sacrifice in overall accuracy.